RoboticonColony/Assets/Editor/Unit_Tests/MarketTest.cs

```
@@ -0,0 +1,173 @@
+
+        //Verify that the markets inventory is empty
+        Assert.AreEqual(0, market.Stock.Money);
+        Assert.AreEqual(0, market.Stock.GetItemAmount(ItemType.Ore));
+        Assert.AreEqual(0, market.Stock.GetItemAmount(ItemType.Power));
+        Assert.AreEqual(0, market.Stock.GetItemAmount(ItemType.Roboticon));
+    }
+
+	[Test]
+	public void CreateMarketWithInventory()
+	{
+        //Create an inventory to give to the market
+        Inventory inv = new Inventory(1, 4, 2, 3);
+
+        //Create the market instance
+        Market market = new Market(inv);
+
+        //Verify that the market contains the items from the inventory
+        Assert.AreEqual(1, market.Stock.Money);
+        Assert.AreEqual(4, market.Stock.GetItemAmount(ItemType.Ore));
+        Assert.AreEqual(2, market.Stock.GetItemAmount(ItemType.Power));
+        Assert.AreEqual(3, market.Stock.GetItemAmount(ItemType.Roboticon));
+    }
+
+    [Test]
+    public void GettingBuyPrices()
+    {
+        //Create an empty market instance
+        Market market = new Market();
+
+        //Verify the buy price is the same as the price set in the markets construction
+        //TEMP: This will likely get set an alternate way in future
+        Assert.AreEqual(10, market.GetBuyPrice(ItemType.Ore));
+        Assert.AreEqual(11, market.GetBuyPrice(ItemType.Power));
+        Assert.AreEqual(12, market.GetBuyPrice(ItemType.Roboticon));
+    }
+
+    [Test]
+    public void GettingSellPrices()
+    {
+        //Create an empty market instance
+        Market market = new Market();
+
+        //Verify the sell price is the same as the price set in the markets construction
+        //TEMP: This will likely get set an alternate way in future
+        Assert.AreEqual(9, market.GetSellPrice(ItemType.Ore));
+        Assert.AreEqual(8, market.GetSellPrice(ItemType.Power));
+        Assert.AreEqual(7, market.GetSellPrice(ItemType.Roboticon));
+    }
+
+    [Test]
+    public void FailedBuy()
```

again i think we should throw an error rather than return false

---

Done